backends/bmv2/parser.cpp

```diff
@@ -286,18 +292,26 @@ ParserConverter::convertSelectExpression(const IR::SelectExpression* expr) {
     for (auto sc : se->selectCases) {
         auto trans = new Util::JsonObject();
         mpz_class value, mask;
-        unsigned bytes = combine(sc->keyset, se->select, value, mask);
-        if (mask == 0) {
-            trans->emplace("value", "default");
-            trans->emplace("mask", Util::JsonValue::null);
+        bool is_vset;
+        cstring vset_name;
+        unsigned bytes = combine(sc->keyset, se->select, value, mask, is_vset, vset_name);
+        if (is_vset) {
+            trans->emplace("value", vset_name);
+            trans->emplace("mask", mask);
             trans->emplace("next_state", stateName(sc->state->path->name));
         } else {
-            trans->emplace("value", stringRepr(value, bytes));
-            if (mask == -1)
+            if (mask == 0) {
```

Comment:
minor nit:  I feel

   } else if (mask == 0) {

and not increasing the indent would be better.